### PR TITLE
Virtual Kubelet documentation

### PR DESCRIPTION
**Pull request by mlavacca**
# Description

This PR is related to the Virtual Kubelet architecture documentation.

### CODE REVIEW

File: docs/pages/Architecture/API sharing/_index.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+title: API sharing
+weight: 4
+---
+
+## Overview
+
+The API sharing in Liqo is implemented by exploiting our implementation of the
```
Comment (palexster) [inline docs/pages/Architecture/API sharing/_index.md:8]: API Sharing?

File: docs/pages/Architecture/API sharing/_index.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Similarly, the virtual kubelet is in charge of:
+* creating the virtual node resource and reconciling its status by taking information from two different resources:
+    * `Advertisement`: node capacity and allocatable.
+    * `TunnelEndpoint`: inter-connectivity parameters.
+* offloading the local pod scheduled on the virtual node to the remote cluster, by creating a remote replicaset that 
+handles the remote pod lifecycle. Besides, it reconciles the local pod status with the remote one, to provide complete visibility
+on the resources offloaded to the remote cluster.
+
+Also, our implementation provides a feature we called "reflection": the offloaded pods may need some resources to be 
+properly executed in the remote cluster, such as services, endpoints, secret, configmaps. The virtual Kubelet itself is
```
Comment (palexster) [inline docs/pages/Architecture/API sharing/_index.md:25]: I would highlight the list of reflected objects as items with *.

**Review by frisso — DISMISSED** [review]

**Comment by frisso** [comment]
@mlavacca Thanks for this work.
Let's make further changes when the other documentation is added.

File: docs/pages/Architecture/_index.md [hunk 3]
```diff
@@ -6,9 +6,9 @@ weight: 2
...
 Liqo enables resource sharing across Kubernetes clusters.
-This section presents the architectural documentation of Liqo, composed of 4 main areas:
+This section presents the architectural documentation of Liqo, composed of four main sections:
 
-* [Discovery](./discovery): automatic detection of other clusters to peer with;
-* [Peering](./peering): resource sharing control plane;
-* [Computing](./computing): resource sharing data plane;
-* [Networking](./networking): interconnection between different clusters.
+* [Discovery](./discovery): how Liqo detects other clusters to peer with;
+* [Peering](./peering): how Liqo exchange information about the resources (and associated costs) available in the other (_foreign_) cluster, and eventually establishing a _peering_ relationship with it;
```
Comment (giorio94) [inline docs/pages/Architecture/_index.md:12]:
```suggestion
* [Peering](./peering): how Liqo exchanges information about the resources (and associated costs) available in the other (_foreign_) cluster, and eventually establishing a _peering_ relationship with it;
```

File: docs/pages/Architecture/offloading/features/computing.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- The pod is deleted in the **foreign** cluster (intentionally or by eviction): not only the local pod must not be deleted, but the remote pod has to be re-created as soon as possible. In fact, the two-steps scheduling process does not allow the foreign cluster to trigger a re-scheduling in the home cluster; the requested pod must be running in the foreign cluster, without involving the home cluster in any un-necessary detail, unless an unrecoverable condition is detected.
+
+To implement this behavior, we have to remember that, in Kubernetes, pods are commonly created starting from a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) or a [Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/).
+However, in Liqo, the request for a new deployment scheduled on a foreign cluster triggers the Liqo controller-manager (running in the virtual kubelet) to create a new [ReplicaSet](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/), that ultimately leads to the creation of the Pod (again, through the controller-manager).
```
Comment (giorio94) [inline docs/pages/Architecture/offloading/features/computing.md:22]:
```suggestion
However, in Liqo, the request for a new deployment scheduled on a foreign cluster triggers the Liqo controller-manager (running in the virtual kubelet) to create a new [ReplicaSet](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/), that ultimately leads to the creation of the Pod (again, through the controller-manager).
```

I suppose this should be rephrased as AFAIU Liqo does not deal with Deployments and ReplicaSets (managed by the vanilla controller-manager), but with the pods created by those.

**Review by giorio94 — DISMISSED** [review]

**Review by palexster — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]

**Pull request merged**